File: Seminar1/java/VendingMachine/src/Domain/HotDrink.java
package Domain;

/*
 * Класс HotDrink представлен как наследник от класса Bottle
 */
public class HotDrink extends Bottle {

    private float cupTemperature;

    public HotDrink(int price, int place, String name, long id, float bottleVolume, float cupTemperature) {
        super(price, place, name, id, bottleVolume);
        this.cupTemperature = cupTemperature;
    }

    public float getCupTemperature() {
        return cupTemperature;
    }

    public void setCupTemperature(float cupTemperature) {
        this.cupTemperature = cupTemperature;
    }

    @Override
    public String toString() {
        // Переопределили вывод для горячих напитков добавлена температура напитка
        return super.toString() + "temperature = " + cupTemperature + " Celsius\n";
    }

}

File: Seminar1/python/VendingMachine/Domain/hotDrink.py
from Domain.product import Product


class HotDrink(Product):
    def __init__(self, price, place, name, id):
        super().__init__(price, place, name, id)
        self.hotDrinkVolume = 0.0

    def __init__(self, price, place, name, id):
        super().__init__(price, place, name, id)

        self.hotDrinkTemperature = 0.0

    def __init__(self, price, place, name, id, hotDrinkVolume, hotDrinkTemperature):
        super().__init__(price, place, name, id)
        self.hotDrinkVolume = hotDrinkVolume
        self.hotDrinkTemperature = hotDrinkTemperature

    def getHotDrinkVolume(self):
        return self.hotDrinkVolume

    def setHotDrinkVolume(self, hotDrinkVolume):
        self.hotDrinkVolume = hotDrinkVolume

    def getHotDrinkTemperature(self):
        return self.hotDrinkTemperature

    def setHotDrinkVolume(self, hotDrinkTemperature):
        self.hotDrinkTemperature = hotDrinkTemperature

    def __str__(self):
        return (
            super().__str__()
            + "volume = "
            + str(self.hotDrinkVolume)
            + "\ntemperature = "
            + str(self.hotDrinkTemperature)
        )
